astropy/io/fits/convenience.py

```diff
@@ -433,15 +434,24 @@ def writeto(filename, data, header=None, output_verify='exception',
                 checksum=checksum)
 
 
-def table_to_hdu(table):
+def table_to_hdu(table, astropy_native=False):
```

Comment:
Agreed with @AustereCuriosity, this should be `False` by default in this PR.  When it comes to FITS we need to be conservative about API changes.

Nevertheless we can discuss this later since I do see the merit of using the proper FITS Time standard when possible.  We could involve stakeholders like @cdeil who already noted that he was happy with a `False` default for gammapy.  We only really need to decide a month or two before 3.0 comes out.